druntime/src/object.d

```diff
@@ -2405,7 +2422,24 @@ const:
         if (flags & MIimportedModules)
         {
             auto p = cast(size_t*)addrOf(MIimportedModules);
-            return (cast(immutable(ModuleInfo*)*)(p + 1))[0 .. *p];
+            auto embeddedImportedModules = (cast(void**)(p + 1))[0 .. *p];
+
+            version(ModuleInfoNeedDereference)
+            {
+                //auto dereferenced = new ModuleInfo*[](embeddedImportedModules.length);
+                import core.memory : pureMalloc;
+                auto dereferenced = (cast(ModuleInfo**)pureMalloc(ModuleInfo.sizeof * embeddedImportedModules.length))[0 .. embeddedImportedModules.length];
+                foreach(offset, mod; cast(ModuleInfo**[])embeddedImportedModules)
+                {
+                    if (mod !is null)
+                        dereferenced[offset] = *mod;
+                }
+                return cast(typeof(return))dereferenced;
+            }
+            else
+            {
+                return cast(typeof(return))embeddedImportedModules;
+            }
```

Comment:
This hack won't do. The manual dllimport relocation necessary at runtime affects *all* imported data symbols; ModuleInfos are just a tiny share of those.

Reply:
How is calling toImport which does the dllimport on importedModules member (and only there), affect other symbols?

Reply:
What I mean is that you need a general solution for *all* data symbols, not a hack which might work for ModuleInfos.

Reply:
I agree.

That is super out of what I can do and unfortunately, I seem to be the only one willing to even give it a go ;)